buzzmobile/drive/car_interface/src/main.cpp

```diff
@@ -0,0 +1,88 @@
+#include <ros/ros.h>
+#include "arduino.h"
+//#include <core_msgs/Odom.h>
+#include <buzzmobile/CarPose.h>
+//#include <core_msgs/MotionCommand.h>
+#include <std_msgs/Bool.h>
+
+using namespace std;
+
+const double ticksPerRev = 3600;
+const double wheelCirc = 2.198;
+
+ros::Publisher encoder_pub;
+ros::Subscriber command_sub;
+ros::Subscriber horn_sub;
+
+Arduino arduino;
+
+//int odom_sequence = 0;
+
+ros::Time last_command_time;
+
+ros::Duration keep_alive_frequency(1.0);
+
+//void odometry_callback(int, float);
+
+//void command_callback(core_msgs::MotionCommand::ConstPtr);
+void command_callback(buzzmobile::CarPose::ConstPtr);
+
+void horn_callback(std_msgs::Bool::ConstPtr);
+
+void keep_alive_callback(const ros::TimerEvent&);
+
+int main(int argc, char **argv) {
+
+  ros::init(argc, argv, "car_interface");
+
+  //arduino.open("/dev/arduino_motor_controller", 9600);
+  arduino.open("/dev/ttyACM0", 9600);
+  //arduino.setOdometryCallback(odometry_callback);
+
+  ros::NodeHandle node_handle;
+
+  last_command_time = ros::Time::now();
+
+  //encoder_pub = node_handle.advertise<core_msgs::Odom>("encoder_odom", 1000);
+
+  command_sub = node_handle.subscribe("car_pose", 1, command_callback);
+  
+  //horn_sub = node_handle.subscribe("car_horn", 1, horn_callback);
+
+  ros::Timer keepAliveTimer = node_handle.createTimer(keep_alive_frequency, keep_alive_callback);
+  ros::spin();
+
+  return 0;
+}
+
+void keep_alive_callback(const ros::TimerEvent&) {
+  ROS_INFO("Keep alive callback called");
+  // 1 sec command frequency required to maintain velocity
+  if((ros::Time::now() - last_command_time) > keep_alive_frequency) {
+    arduino.setSpeed(0);
+  }
+}
+/**
+void odometry_callback(int tickCount, float steeringAngle) {
+  core_msgs::Odom msg;
+  msg.distance_travelled = (tickCount * wheelCirc) / ticksPerRev;
+  msg.steering_angle = steeringAngle;
+  msg.header.seq = odom_sequence++;
+  msg.header.stamp = ros::Time::now();
+  encoder_pub.publish(msg);
+}*/
```

Comment:
Just a note that I'm okay with the commented out stuff on this file since there's an open issue for adding this in the future #41 #66